cmd/apiserver-boot/boot/util/signal/doc.go

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Copied from https://github.com/kubernetes/apiserver/tree/729b73dd4aed4c03c6757f1c453a317578a8e1f0/pkg/server
+
+TODO: Remove this copied package and import from k8s.io/apiserver/pkg/server
```

Comment:
can you expand why copy the package from k8s.io/apiserver?

Reply:
I originally attempted to import it, but the dependencies of k8s.io/apiserver clash with the dependencies in this project.
See https://travis-ci.org/kubernetes-sigs/apiserver-builder-alpha/builds/578741517
```
/build-tools.sh

# sigs.k8s.io/apiserver-builder-alpha/cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:181:55: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:182:20: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:229:52: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:230:11: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:237:11: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:245:11: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:252:12: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:263:12: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:265:40: undefined: schema.UnionField

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:272:12: undefined: schema.Union

../../cmd/vendor/k8s.io/kube-openapi/pkg/schemaconv/smd.go:272:12: too many errors
```

Reply:
yea, but i updated the vendor by https://github.com/kubernetes-sigs/apiserver-builder-alpha/pull/411.. but it's supposed to be working

Reply:
@yue9944882 Thanks, but sadly it still doesn't compile. I now get:

```
richard   717a61c  …  src  sigs.k8s.io  apiserver-builder-alpha  130  make install
rm -rf *.deb *.rpm *.tar.gz ./release
go run ./cmd/apiserver-builder-release/main.go vendor --version v1.15.alpha.0 --commit 717a61c5e9922a63741934090526756d87db6186
cp -R -H vendor /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/src
cp -R -H pkg /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/src/vendor/sigs.k8s.io/apiserver-builder-alpha/pkg
bash -c find /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/src/vendor/sigs.k8s.io/apiserver-builder-alpha/pkg -name BUILD.bazel| xargs sed -i='' s'|//pkg|//vendor/sigs.k8s.io/apiserver-builder-alpha/pkg|g'
cp -R -H Gopkg.toml /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/src/Gopkg.toml
cp -R -H Gopkg.lock /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/src/Gopkg.lock
go run ./cmd/apiserver-builder-release/main.go build --version v1.15.alpha.0
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/gen-apidocs cmd/vendor/github.com/kubernetes-incubator/reference-docs/gen-apidocs/main.go
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/client-gen cmd/vendor/k8s.io/code-generator/cmd/client-gen/main.go
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/conversion-gen cmd/vendor/k8s.io/code-generator/cmd/conversion-gen/main.go
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/deepcopy-gen cmd/vendor/k8s.io/code-generator/cmd/deepcopy-gen/main.go
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/defaulter-gen cmd/vendor/k8s.io/code-generator/cmd/defaulter-gen/main.go
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/informer-gen cmd/vendor/k8s.io/code-generator/cmd/informer-gen/main.go
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/lister-gen cmd/vendor/k8s.io/code-generator/cmd/lister-gen/main.go
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/openapi-gen cmd/vendor/k8s.io/kube-openapi/cmd/openapi-gen/openapi-gen.go
go build -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/apiregister-gen cmd/apiregister-gen/main.go
go build -ldflags -X sigs.k8s.io/apiserver-builder-alpha/cmd/apiserver-boot/boot/version.apiserverBuilderVersion=v1.15.alpha.0 -X sigs.k8s.io/apiserver-builder-alpha/cmd/apiserver-boot/boot/version.kubernetesVendorVersion=1.15.3 -X sigs.k8s.io/apiserver-builder-alpha/cmd/apiserver-boot/boot/version.goos=linux -X sigs.k8s.io/apiserver-builder-alpha/cmd/apiserver-boot/boot/version.goarch=amd64 -X sigs.k8s.io/apiserver-builder-alpha/cmd/apiserver-boot/boot/version.gitCommit=717a61c5e9922a63741934090526756d87db6186
 -X sigs.k8s.io/apiserver-builder-alpha/cmd/apiserver-boot/boot/version.buildDate=2019-08-30-12:44:30 -o /home/richard/go/src/sigs.k8s.io/apiserver-builder-alpha/release/v1.15.alpha.0/bin/apiserver-boot cmd/apiserver-boot/main.go
# sigs.k8s.io/apiserver-builder-alpha/cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/gvkparser.go:41:3: cannot use nil as type typed.ParseableType in return argument
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:45:14: cannot use make(map[string]*fieldpath.VersionedSet) (type map[string]*fieldpath.VersionedSet) as type fieldpath.ManagedFields in return argument
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:78:16: cannot use make(map[string]*fieldpath.VersionedSet, len(encodedManagedFields)) (type map[string]*fieldpath.VersionedSet) as type fieldpath.ManagedFields in assignment
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:84:31: cannot assign *fieldpath.VersionedSet to managedFields[manager] (type fieldpath.VersionedSet) in multiple assignment:
	*fieldpath.VersionedSet is pointer to interface, not interface
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:116:17: invalid pointer type *fieldpath.VersionedSet for composite literal
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:117:14: versionedSet.APIVersion undefined (type *fieldpath.VersionedSet is pointer to interface, not interface)
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:119:15: versionedSet.Applied undefined (type *fieldpath.VersionedSet is pointer to interface, not interface)
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:130:14: versionedSet.Set undefined (type *fieldpath.VersionedSet is pointer to interface, not interface)
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:147:38: cannot use versionedSet (type fieldpath.VersionedSet) as type *fieldpath.VersionedSet in argument to encodeManagerVersionedSet:
	*fieldpath.VersionedSet is pointer to interface, not interface
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/versionconverter.go:35:5: cannot use &versionConverter literal (type *versionConverter) as type merge.Converter in assignment:
	*versionConverter does not implement merge.Converter (wrong type for Convert method)
		have Convert(typed.TypedValue, fieldpath.APIVersion) (typed.TypedValue, error)
		want Convert(*typed.TypedValue, fieldpath.APIVersion) (*typed.TypedValue, error)
cmd/vendor/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go:147:38: too many errors
2019/08/30 12:44:32 exit status 2
exit status 1
make: *** [Makefile:62: build] Error 1

```

Reply:
what's commit 717a61c5e? it doesn't exist in the repo

Reply:
That's a local commit, where I'm in the process of rebasing on top of master. 
I'll force push it so that you can see the build results.

Reply:
thanks, i re-pro'd locally, now the build is passing